src/AppInstallerRepositoryCore/Rest/RestSourceFactory.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "Public/AppInstallerRepositorySource.h"
+#include "SourceFactory.h"
+#include <string_view>
+
+namespace AppInstaller::Repository::Rest
+{
+    using namespace std::string_view_literals;
+
+    // A source where the information is stored on a REST based server.
+    // In addition, the manifest information is also available on the server.
+    // Arg  ::  Expected to be a API which supports querying functionality.
+    struct RestSourceFactory
+    {
+        // Get the type string for this source.
+        static constexpr std::string_view Type()
+        {
+            using namespace std::string_view_literals;
+            return "ThirdParty.RestSource"sv;

```

Comment:
I don't think we should call it "ThirdParty", as Microsoft will certainly implement and use these sources.

I would prefer `Microsoft.Rest` as the API definition is ours.